src/components/options/providers/groups.svelte

```diff
@@ -1,41 +1,61 @@
 <script>
+import _ from "lodash";
 import Notiflix from "notiflix";
 import { createEventDispatcher } from "svelte";
 
 import LocalStore from "../../../js/shared/local_store";
 import { StoreKey } from "../../../js/shared/constants";
 
-// Props.
-export let initialSettings;
+const dispatch = createEventDispatcher();
+
+// Auxiliary variable to store initial settings.
+let initialSettings;
 
+// States.
+let settings;
```

Comment:
Not keeping and copy of `seetings` and doing:

```
const settings = await LocalStore.getOne(StoreKey.SETTINGS);
settings.providersGroups = newGroups;
await LocalStore.setOne(StoreKey.SETTINGS, settings);
```

Every time you want to save a change is intended, since this isn't the only component writing on the local store.

For example:

1. User opens the page, `groups.svelte` is mounted and `settings` is initialized with `{ x: 1, providerGroups: [a,b,c] }`.
2. User clicks on another component that changes `x` to 2, so new settings are `{ x: 2, providerGroups: [a,b,c] }`.
3. User changes the name of a group from `c` to `d`, so `groups.svelte` writes `{ x: 1, providerGroups: [a,b,d] }`, undoing the changes on `x` because `groups.svelte` made a copy of the `settings` before `x` was changed.